xla/service/algorithm_util.cc

```diff
@@ -155,11 +155,17 @@ bool IsSupportedDotAlgorithmOnGpu(
       std::get<se::CudaComputeCapability>(gpu_compute_capability)
           .IsAtLeast(8, 9);
 
+  const bool is_rocm_mi100_and_above =
+      std::holds_alternative<se::RocmComputeCapability>(
+          gpu_compute_capability) &&
+      std::get<se::RocmComputeCapability>(gpu_compute_capability)
+          .gfx9_mi100_or_later();
+
   switch (algorithm) {
     case PrecisionConfig::ALG_DOT_ANY_F8_ANY_F8_F32:
     case PrecisionConfig::ALG_DOT_ANY_F8_ANY_F8_F32_FAST_ACCUM:
       // Other F8 types are actually not supported by NVIDIA GPUs.
-      return is_cuda_ge_ada &&
+      return (is_cuda_ge_ada || is_rocm_mi100_and_above) &&
```

Comment:
I am not sure if this is really correct. I guess FP8 support begins from MI300 arch? But I see that we have the same check also on upstream